src/bit-systems/network-receive-system.ts

```diff
@@ -146,27 +198,24 @@ export function networkReceiveSystem(world: HubsWorld) {
       Networked.lastOwnerTime[eid] = updateMessage.lastOwnerTime;
       Networked.timestamp[eid] = updateMessage.timestamp;
 
-      // TODO HACK simulating a buffer with a cursor using an array
-      updateMessage.data.cursor = 0;
-      for (let s = 0; s < updateMessage.componentIds.length; s++) {
-        const componentId = updateMessage.componentIds[s];
-        const schema = schemas.get(networkableComponents[componentId])!;
-        schema.deserialize(world, eid, updateMessage.data);
+      if (isCursorBufferUpdateMessage(updateMessage)) {
```

Comment:
I wonder if these different intuitions are because of a difference of opinion on how we should handle the `StorableUpdateMessage`s. 

You had expressed a preference for first transforming them into `CursorBufferUpdateMessage`s beforehand, and then applying them the same way as any others. That has the advantage of keeping the `networkReceiveSystem` very simple: It only deals with one update message shape. 

The cost is that we have to be able to write the transformer from JSON to a `CursorBufferUpdateMessage`. This seems more difficult than applying the JSON to a `NetworkedFoo` component, so I opted to add the complexity to the network receive system. 

This is intertwined with however we decide to do serialization / storage so I expect we'll do some iteration here.

As for removing it -- I'm a little hesitant because I feel like it will leave the storage / serialization code especially incomplete / incoherent. (Removing pinning seems OK because we don't want to write to the DB yet, but removing these lines makes this initial attempt at serialization / storage harder to understand and reason about.)